src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -216,9 +214,17 @@ public static void AddExceptionInformation(in WriteStack writeStack, JsonExcepti
             ex.Path = path;
 
             // If the message is empty, use a default message with the Path.
-            if (string.IsNullOrEmpty(ex.Message))
+            string message = ex.Message;
+            if (string.IsNullOrEmpty(message))
             {
-                ex.SetMessage(SR.Format(SR.SerializeUnableToSerialize, path));
+                message = SR.Format(SR.SerializeUnableToSerialize);
+                ex.AppendPathInformation = true;
+            }
+
+            if (ex.AppendPathInformation)
```

Comment:
Question: Why do we do it this way, rather than always appending the path information (like we had before)?

Reply:
It mirrors the other code for ReadStack which allows a caller to set `AppendPathInformation=true`  (although currently no code does that here for WriteStack).